core/commands/pin.go

```diff
@@ -364,7 +424,7 @@ new pin and removing the old one.
 	Options: []cmdkit.Option{
 		cmdkit.BoolOption(pinUnpinOptionName, "Remove the old pin.").WithDefault(true),
 	},
-	Type: PinOutput{},
+	Type: PinUpdateOutput{},
```

Comment:
While this objectively makes more sense, it's also api-breaking. There are some loose plans to implement 'next-gen' api - https://github.com/ipfs/http-api-spec/issues/116 which should address issues like this, but until this happens we'd like to keep this api as stable as possible,